### DISCUSSION TITLE
Unable to upgrade only a selected NewGRF in active GRF list

### DISCUSSION THREAD

**Discussion by LC-Zorg** [OpenTTD/OpenTTD#11830] [Features]
## Version of OpenTTD
1.10.3
## Expected result
By selecting an add-on in the NewGRF active list, the update will only apply to that add-on.
## Actual result
By selecting an add-on on the NewGRF list of active add-ons, all add-ons are updated, including those that the player doesn't want to update.
## Steps to reproduce
1. Select several older versions of the add-ons to the list of active add-ons
2. Try to upgrade one add-on

**Comment by LordAro** [comment]
I'll try to put this more plainly:

* Activate some NewGRFs
* Go to content download
* Notice that multiple of your now-active NewGRFs are out of date
* Update them
* Go back to your active NewGRFs window
* Select a single active NewGRF
* Press 'Upgrade'
* Notice how all out of date NewGRFs have now been updated, rather than just the one you selected

I suspect this is a case of "working as intended", rather than anything else (after all, why would you want to use any out of date NewGRF?) and just the UI being unclear that selection has no effect here - the Upgrade button is active without selecting anything, after all

**Comment by LC-Zorg** [comment]
> why would you want to use any out of date NewGRF?

Some earlier versions are just better. :) An example would be the CZTR Tree set. For me, the first version looks much better. Sometimes they also have interesting features that were removed in newer versions or on the contrary, they are not overloaded with features . 
When I have a long preset and only want to update one add-on, I can't use "upgrate" because that will change all other add-ons, I have to search for it in the list, manually replace it and then change the settings after that. It is a bit tiresome when I testing an add-on and I comparing older versions with newer ones.
I admit that this may not be a very common problem. I reported it by the way. Perhaps it can be solved by the occasion. :)
